static/app/components/globalDrawer/components.tsx

```diff
@@ -0,0 +1,97 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import type {DrawerOptions} from 'sentry/components/globalDrawer';
+import {IconClose} from 'sentry/icons/iconClose';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import useKeyPress from 'sentry/utils/useKeyPress';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+import SlideOverPanel from 'sentry/views/insights/common/components/slideOverPanel';
+
+export interface DrawerPanelProps extends DrawerOptions {
+  children: React.ReactNode;
+  isOpen?: boolean;
+}
+
+export function DrawerPanel({
+  children,
+  isOpen = false,
+  onClose,
+  onOpen,
+  closeOnOutsideClick = true,
+  closeOnEscapeKeypress = true,
+}: DrawerPanelProps) {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(isOpen);
+  const escapeKeyPressed = useKeyPress('Escape');
+  const panelRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside(panelRef, () => {
+    if (isOpen && closeOnOutsideClick) {
+      onClose?.();
+      setIsDrawerOpen(false);
+    }
+  });
+
+  useEffect(() => {
+    if (isOpen && escapeKeyPressed && closeOnEscapeKeypress) {
+      onClose?.();
+      setIsDrawerOpen(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [escapeKeyPressed, closeOnEscapeKeypress]);
+
+  useEffect(() => {
+    setIsDrawerOpen(isOpen);
+  }, [isOpen]);
+
+  return (
+    <div role="complementary" aria-hidden={!isDrawerOpen}>
```

Comment:
Seemed like a good and likely unused accessibility role  - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role